ui-lib/src/main/java/co/electriccoin/zcash/ui/common/datasource/LocalAddressBookDataSource.kt

```diff
@@ -59,14 +72,15 @@ class LocalAddressBookDataSourceImpl(
 
     override suspend fun saveContact(
```

Comment:
Non-blocking:

This will write the address book file on every update to a contact. Are there any situations currently where we update multiple contacts at once? If there are then we should probably have a batch update API.

Reply:
Currently we do not have multiple contacts CRUD operations at once.

Also we do support have parallel CRUD operations - instead we implemented a semaphore and the operations are executed in the order in which they were requested.